### PR TITLE
API review changes

### PR DESCRIPTION
**Pull request by roji**
Part of #33220

### CODE REVIEW

File: src/EFCore.Relational/Query/QuerySqlGenerator.cs [hunk 1]
```diff
@@ -1029,25 +1093,25 @@ protected override Expression VisitAtTimeZone(AtTimeZoneExpression atTimeZoneExp
...
     /// </summary>
-    /// <param name="selectExpression">A select expression to use.</param>
+    /// <param name="selectExpression">The <see cref="SelectExpression" /> for whose TOP clause to generate SQL.</param>
     protected virtual void GenerateTop(SelectExpression selectExpression)
     {
     }
 
     /// <summary>
-    ///     Generates the projection in the relational command
+    ///     Generates SQL for the projection clause of the given SELECT expression.
```
Comment (roji) [inline src/EFCore.Relational/Query/QuerySqlGenerator.cs:1112]: [The API review](https://github.com/dotnet/efcore/issues/33220) has a note to rename GenerateProjection to GenerateFrom, which I think is a confusion with GenerateTables (I did this). I'm not sure what we'd rename GenerateProjection to - it really is the projection of the SELECT (i.e. the part after SELECT and before FROM).

**Review by ajcvickers — APPROVED** [review]

**Comment by roji** [comment]
As the internals analyzer graciously reminded me, we do use SqlAliasManager.GenerateTableAlias() in providers (e.g. [SQL Server](https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Query/Internal/SqlServerQueryableMethodTranslatingExpressionVisitor.cs#L332), e.g. when a JSON lookup is transformed into OPENJSON by composing a LINQ operator on a JSON collection. So making SqlAliasManager fully pubternal would mean using internal APIs from providers.

We can at least keep SqlAliasManager (and SqlAliasManagerFactory?) full-public because of this, but only publicly expose APIs that are for provider consumption (as opposed to APIs for overriding/customizing actual alias logic).

Will bring to design.

**Review by roji — COMMENTED** [review]
I now remember why I didn't include a check to ensure no aliases are generated after the postprocessing pass has occured... We occasionally need to generate new aliases in SqlNullabilityProcessor; that's in the 2nd part of the query pipeline, and so after post-processing. I debated while implementing if the alias postprocessing run should be moved to the end of the 2nd part of the query pipeline, but we want to keep that as light as possible, and alias post-processing is inherently optional (close gaps).

So for now I'll leave it as it is until we have a good reason to do otherwise (if anyone objects, let me know and we'll revisit).

**Comment by AndriySvyryd** [comment]
> So for now I'll leave it as it is until we have a good reason to do otherwise (if anyone objects, let me know and we'll revisit).

Just add a comment there, so we take it into account in the future

**Pull request merged**